app/helpers/application_helper/button/basic.rb

```diff
@@ -42,9 +42,15 @@ def localized(key, value = nil)
   #   self[:prompt]  -- the user has to confirm the action
   #   self[:hidden]  -- the button is not displayed in the toolbar
   #   self[:text]    -- text for the button
+  #   self[:onwhen]  -- enable button if items are selected
   def calculate_properties
-    self[:enabled] = !disabled? if self[:enabled].nil?
-    self[:title] = @error_message if @error_message.present?
+    enabled = !disabled?
```

Comment:
The changes are nice! But the code is still a bit unreadable, can you add some comments here?

`# returns enabled state and sets @error_message`